io-engine/src/subsys/nvmf/admin_cmd.rs

```diff
@@ -158,15 +158,15 @@ pub fn create_snapshot(
             | (nvme_cmd_cdw11_get_val(cmd) as u64) << 32
     };
     let snapshot_name = Lvol::format_snapshot_name(&lvol.name(), snapshot_time);
-    let snap_param = SnapshotParams::new(
+    let mut snap_param = SnapshotParams::new(
         Some(lvol.name()),
         Some(lvol.name()),
         Some(lvol.name()),
         Some(snapshot_name),
     );
     // Blobfs operations must be on md_thread
     Reactors::master().send_future(async move {
-        match lvol.create_snapshot(snap_param).await {
+        match lvol.create_snapshot(&mut snap_param).await {
```

Comment:
I'm curious, why does it need to be a reference rather than moved?

Reply:
> I'm curious, why does it need to be a reference rather than moved?

Currently the idea is as part of create_snapshot, some of the `snapshotparams` will be set as part of SPDK call back function. To get the changed parameter information in gRPC code (replica.rs), I made it mutable reference.

Reply:
@hrudaya21 Sometimes it makes to pass by value (move) and return it. Like this:
`
fn foo(a: Data) -> Data { a.x = 123; a }

let a = Data::new();
let a = foo(a);
`

instead of 

`
fn foo(a: &mut Data) { a.x = 123; }

let a = Data::new();
foo(&mut a);
`

Mut references are very restrictive in Rust, for very good reasons. By moving data, you avoid having mut ref.

Reply:
understood. currently there is no requirement for changing the param though. But understood the rust best practice and modified code.